lib/problems.rb

```diff
@@ -1,13 +1,66 @@
-require_relative './stack.rb'
+require_relative 'stack'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(N)
+# Space Complexity: O(N)
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  strStack = Stack.new
+  string.each_char do |char|
+    if ['(', '[', '{'].include?(char)
+      strStack.push(char)
+    else
+      element = strStack.pop 
+      if char == ']' && element != '['
+        return false
+      elsif char == ')' && element != '('
+        return false
+      elsif char == '}' && element != '{'
+        return false  
+      end
```

Comment:
This could be dried up using a hash with keys being the open brace and the value being the corresponding close brace.